File: frontend/src/app/meeting/services/meeting.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Meeting, MeetingStatus } from '../models/meeting.model';
import { User } from '../../user/models/user.model';
import { environment } from '../../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class MeetingService {
  private apiUrl = `${environment.apiUrl}/meetings`;

  constructor(private http: HttpClient) { }

  createMeeting(meeting: Partial<Meeting>): Observable<Meeting> {
    return this.http.post<Meeting>(this.apiUrl, meeting);
  }

  updateMeeting(id: number, meeting: Partial<Meeting>): Observable<Meeting> {
    return this.http.put<Meeting>(`${this.apiUrl}/${id}`, meeting);
  }

  deleteMeeting(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  getMeetingById(id: number): Observable<Meeting> {
    return this.http.get<Meeting>(`${this.apiUrl}/${id}`);
  }

  getAllMeetings(): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(this.apiUrl);
  }

  getMeetingsByOrganizer(userId: number): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(`${this.apiUrl}/organizer/${userId}`);
  }

  getMeetingsByParticipant(userId: number): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(`${this.apiUrl}/participant/${userId}`);
  }

  getMeetingsByDateRange(startDate: Date, endDate: Date): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(`${this.apiUrl}/date-range`, {
      params: {
        startDate: startDate.toISOString(),
        endDate: endDate.toISOString()
      }
    });
  }

  getUpcomingMeetings(): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(`${this.apiUrl}/upcoming`);
  }

  getOverdueMeetings(): Observable<Meeting[]> {
    return this.http.get<Meeting[]>(`${this.apiUrl}/overdue`);
  }

  addParticipant(meetingId: number, userId: number): Observable<Meeting> {
    return this.http.put<Meeting>(`${this.apiUrl}/${meetingId}/participants/add`, { userId });
  }

  removeParticipant(meetingId: number, userId: number): Observable<Meeting> {
    return this.http.put<Meeting>(`${this.apiUrl}/${meetingId}/participants/remove`, { userId });
  }

  markAsCompleted(meetingId: number): Observable<Meeting> {
    return this.http.put<Meeting>(`${this.apiUrl}/${meetingId}/status`, { status: MeetingStatus.COMPLETED });
  }

  updateMeetingStatus(meetingId: number, status: MeetingStatus): Observable<Meeting> {
    return this.http.put<Meeting>(`${this.apiUrl}/${meetingId}/status`, { status });
  }
} 

File: backend/src/main/java/com/tunisair/meetingmanagement/service/MeetingService.java
package com.tunisair.meetingmanagement.service;

import com.tunisair.meetingmanagement.model.Meeting;
import com.tunisair.meetingmanagement.model.User;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.Set;

public interface MeetingService {
    List<Meeting> getAllMeetings();
    Optional<Meeting> getMeetingById(Long id);
    Meeting createMeeting(Meeting meeting);
    Meeting updateMeeting(Long id, Meeting meetingDetails);
    void deleteMeeting(Long id);
    List<Meeting> getMeetingsByCreator(User user);
    List<Meeting> getMeetingsByParticipant(User user);
    List<Meeting> getMeetingsByDateRange(LocalDateTime start, LocalDateTime end);
    List<Meeting> searchMeetingsByTitle(String title);
    Set<Meeting> getMeetingsByParticipantId(Long userId);
    Meeting addParticipant(Long meetingId, Long userId);
    Meeting removeParticipant(Long meetingId, Long userId);
} 